woocommerce/woocommerce-ios
# [Mobile Payments] Users can be reminded later for feature announcement cards

Closes: #7252 


⛔ Do not merge until b62353706c0cb0c8f93254e7c402ebbecd983f30 is reverted: This is just to make testing easier.

### Description


Feature Announcement Cards can be dismissed by the user. Some users may wish to temporarily get a card out of the way, but be interested in acting on it in future.

This adds a choice for the user when dismissing a card: “Don’t show again”, or “Remind me later”.

A user who chooses “Remind me later” will be shown the card again after 14 days.

### Testing instructions


To help with testing, I recommend changing [this line in AppSettingsStore](https://github.com/woocommerce/woocommerce-ios/blob/54069d4ff84ae1e590e9a4598408dbd19b2c7f79/Yosemite/Yosemite/Stores/AppSettingsStore.swift#L743) to the following, which will change the remindAfter time from 2 weeks to 20 seconds:
```
let remindAfter = remindLater ? Date().addingTimeInterval(20) : nil  // 20 seconds for testing!
```

Using a US or Canada store:

1. Navigate to Order Details for an unpaid order, then tap Collect Payment. Or: make a new Simple Payment, and go to the Payment Method screen
2. Observe that the Upsell Card Reader banner is shown, with a x button to dismiss.
3. Tap the x button and then the `Remind me later` option – observe that the card is dismissed
4. Go to another order within the remind time, and observe that the banner doesn't show there.
5. Close the order an go to any other order after the remind time elapses, observe that the card is shown.

Repeat the above test using `Don't show again`, and observe that the card is not shown even after the remind time elapses.

To reset and re-test, either delete and reinstall the app, or use the testing workaround from b62353706c0cb0c8f93254e7c402ebbecd983f30 to clear the stored dismissal. This is not part of the feature, and will be removed before it is merged.

1. Tap the `Card` option on the Payment Method screen to reset the stored preference.
2. Close the order, and open it or any other order again
3. Observe that the banner is shown

### Screenshots


https://user-images.githubusercontent.com/2472348/178279172-ce776c45-612e-4516-b4cd-152483f6cda6.mp4




---
